pkg/core/statesync/neotest_test.go

```diff
@@ -22,262 +22,413 @@ func TestStateSyncModule_Init(t *testing.T) {
-			count--
-			if count < 0 {
-				break
-			}
-		}
 
-		// then create new statesync module with the same DB and check that state is proper
-		// (headers are in sync, mpt is not yet synced)
-		module = bcBolt.GetStateSyncModule()
-		require.NoError(t, module.Init(bcSpout.BlockHeight()))
-		require.True(t, module.IsActive())
-		require.True(t, module.IsInitialized())
-		require.False(t, module.NeedHeaders())
-		require.True(t, module.NeedStorageData())
-		require.False(t, module.NeedBlocks())
-		unknownNodes = module.GetUnknownMPTNodesBatch(100)
-		require.True(t, len(unknownNodes) > 0)
-		require.NotContains(t, unknownNodes, expectedHeader.PrevStateRoot)
-		require.Panicsf(t, func() { module.BlockHeight() }, "block height is not yet initialized since MPT is not in sync")
-
-		// add the rest of MPT nodes and check that MPT is in sync
-		alreadyRequested := make(map[util.Uint256]struct{})
-		for {
-			unknownHashes := module.GetUnknownMPTNodesBatch(1) // restore nodes one-by-one
-			if len(unknownHashes) == 0 {
-				break
-			}
-			if _, ok := alreadyRequested[unknownHashes[0]]; ok {
-				t.Fatal("bug: node was requested twice")
-			}
-			alreadyRequested[unknownHashes[0]] = struct{}{}
-			var callbackCalled bool
-			err := bcSpout.GetStateSyncModule().Traverse(unknownHashes[0], func(node mpt.Node, nodeBytes []byte) bool {
-				require.NoError(t, module.AddMPTNodes([][]byte{bytes.Clone(nodeBytes)}))
-				callbackCalled = true
-				return true // add nodes one-by-one
+			t.Run("inactive: bolt chain height is close enough to spout chain height", func(t *testing.T) {
+				bcBolt, _, _ := chain.NewMultiWithCustomConfig(t, cc.cfg)
+				for i := uint32(1); i < bcSpout.BlockHeight()-stateSyncInterval; i++ {
+					b, err := bcSpout.GetBlock(bcSpout.GetHeaderHash(i))
+					require.NoError(t, err)
+					require.NoError(t, bcBolt.AddBlock(b))
+				}
+				module := bcBolt.GetStateSyncModule()
+				require.NoError(t, module.Init(bcSpout.BlockHeight()))
+				require.False(t, module.IsActive())
+				require.Equal(t, bcBolt.GetConfig(), module.GetConfig())
 			})
-			require.NoError(t, err)
-			require.True(t, callbackCalled)
-		}
-		unknownNodes = module.GetUnknownMPTNodesBatch(2)
-		require.Equal(t, 0, len(unknownNodes))
 
-		// add several blocks to create DB state where blocks are not in sync yet, but it's not a genesis.
-		for i := stateSyncPoint - maxTraceable + 1; i <= stateSyncPoint-stateSyncInterval-1; i++ {
-			block, err := bcSpout.GetBlock(bcSpout.GetHeaderHash(i))
-			require.NoError(t, err)
-			require.NoError(t, module.AddBlock(block))
-		}
-		require.True(t, module.IsActive())
-		require.True(t, module.IsInitialized())
-		require.False(t, module.NeedHeaders())
-		require.False(t, module.NeedStorageData())
-		require.True(t, module.NeedBlocks())
-		require.Equal(t, uint32(stateSyncPoint-stateSyncInterval-1), module.BlockHeight())
+			t.Run("error: bolt chain is too low to start state sync process", func(t *testing.T) {
```

Comment:
Ditto. What I need for TestStateSyncModule_Init is to extend only those part that is related to contract storage sync initialization, that's it. We need to cover newly-added code of `(s *Module) Init`. The rest part of this test should be kept unchanged, I expect as less changes in `git diff` as possible for this test.